### DISCUSSION TITLE
Steel Reinforcement in SAF

### DISCUSSION THREAD

**Discussion by sonkaste17** [StructuralAnalysisFormat/StructuralAnalysisFormat-Doc#24] [Ideas]
Here we can share ideas on the topic of how to store reinforcement in SAF.

- How to define it for 1D members (beams and columns)?
- How to define it for 2D members (slabs and walls)?
- What all attributes do we need to specify reinforcement properly?

**Comment by robinblazek** [comment]
Hi all, 

we would like to open Reinforcement objects in SAF topic for discussion. As we all know, this topic is quite huge one and can be really complex. There are a lot of possibilities, objects and approaches that can be involved. **That’s why we need your ideas, remarks and opinions more than ever!**
At this first phase, we would like to make it as simple as possible, but at the same time so that there is a possibility of expansion in the future. 
We started with **reinforcement for 1D members**, which basically means 2 main objects – longitudinal rebars and stirrups. 

**Sample of headers for longitudinal rebars**
<img width="1109" alt="image" src="https://user-images.githubusercontent.com/88313526/156178295-ff31a51b-bf02-4485-9251-913e99f7471c.png">

**Explanation of the properties**
| Property | Description |
|---|---|
| Name | Name of the item (rebar) |
| Member | 1D member in which the reinforcement is located |
| Material | Material of the rebars |
| Diameter | Diameter of the rebars |
| Count of bars | Count of bars in one height with the same characteristics, so one line in SAF can describe far more than one bar, it can be seen as a group of bars |
| Offset | Offset of the bar(s) measured from the edge of the cross-section; it allows user to define more than one group of bars in the same height and define their position so that they are not in collision |
| Position Z [mm] | Define Z coordination of the position of bars |
| Coordinate definition | Define if the position is relative to length of the 1D member or it is absolute value |
| Start point [m] | X coordinate of the start point
| End point [m] | X coordinate of the end point
| Master stirrup | Reference to related stirrup (it will be optional – some SW solutions are using it, some don’t) |
| Id | Unique ID of the item |

**Sample of headers for stirrups**
![image](https://user-images.githubusercontent.com/88313526/156178640-481d731a-3163-4c28-b3f5-e33968f31dc8.png)

**Explanation of the properties**
| Property | Description |
|---|---|
| Name | Name of the item (stirrup) |
| Material | Material of the stirrup |
| Cover [mm] | Define offset of the stirrup from the edges of the CSS |
| Diameter | Diameter of the stirrups |
| Count | Count of stirrups in this group; one of the possible ways how to define density of the items: range + count |
| Spacing [mm] | Spacing between each stirrup; alternate way how to define density of the items: range + spacing |
| Type | Define shape of the stirrup; there probably will be also some “general” type with reference to polyline definition of the specific shape |
| Coordinate definition | Define if the position is relative to length of the 1D member or it is absolute value |
| Start point [m] | X coordinate of the start point of the range for stirrup |
| End point [m] | X coordinate of the end point of the range for stirrup |

**Scheme of cross-section with selected properties**
<img width="458" alt="image" src="https://user-images.githubusercontent.com/88313526/156180357-bbab3d66-3cd1-4312-81bd-95d8263a528c.png">


There are a lot of parts to discuss: 

•	How to define Z position of the bars? From bottom/top edge of the CSS or from the centre of gravity? 
•	How to deal with offset, when the edges of CSS will be not perpendicular, for example rounded, or just with inclined edges? Quite similar question is related to “cover” for stirrup. 
•	Big topic is complete 3D geometry of bar itself. We are analysing various possibilities, how to deal with bars, that aren’t straight. One of the proposals is to use separate field/sheet with points defining polyline in 3D dimension. Something similar what we use in SAF for general shape of cross-section (CompositeShapeDef). Do you have any ideas or preferred requirements for this issue? 
•	We also noticed some suggestions about anchorage. We definitely agree that anchorage is quite important part of the reinforcement, but as we tried to make it as simple as possible (at least in this first phase), we are more inclined to leave this property out for now. Do you have different opinion? Is data about anchorage necessary important for you? 
•	There was also idea to have there possibility to mark bars with some specific label, which could be used for identifying the same (geometrically) items in the model, as it is common in bills of materials. Do you find this option interesting for you and your customers? 

Are there any other necessary properties, that you would like to include for the objects? Is this proposal somehow acceptable or do you have completely different opinion, how it should look like? 

**In any case, please let us know, what you think!** As I mentioned at the beginning, this topic is complex and we know, that on the market are couple of different approaches. We would really like to design it in a way, which will suit your solutions and customers.

Thank you all in advance!

**Comment by robinblazek** [comment]
Hi all,

we want to introduce you **the updated version (still draft) of reinforcement in SAF** objects for 1D members. 
After some discussions and your remarks, we made couple of changes. You can see this proposal as a bare minimum of what is really necessary. We think that this is the best way to go - start first with the basic properties and in the next versions build on top of that.

We attach [here](https://scia.sharepoint.com/:x:/s/Calatrava/EfStNLI4kfhAtkS9TWua-tcBt5QjgMzQPllqwcXLz7Cftg?e=ppw5H5) link to SAF, where you can see the proposal. In the sheet _Documentation_ you find all the descriptions and definitions for each property. The sheet _Reinf1DStirrup_ is for stirrups data and _Reinf1DLong_ for longitudinal bars. 

Feel free to download it and use it for testing purposes, you can try manually fill the cells, if it is feasible to define examples you think are important for you and your customers. We want to hear your thoughts and remarks on this proposal. We are now testing this version on real examples. We would like to move forward in coming weeks and start implementation of proof of concept. 

Of course, there are still lot of topics open, so definitely don't hesitate to comment on them too: 
* Longitudinal bars - anchorage
  + Is it needed?
  + Which parameters do you need - length, type, angle, something else?
* Stirrups - detailed definition of the geometry
  + How much detailed it has to be? 
  + Is information about closure/anchorage needed?
  + Would you prefer some format inspired by BVBS (lengths and angles) or just defining polygon by coordinates (for example as in [CompositeShapeDef](https://www.saf.guide/en/stable/structural-analysis-elements/compositeshapedef.html) object in SAF)?
* Longitudinal bars - detailed definition of specific shapes
  + Do we need to cover the cases with different shapes of the bars? 
  + Is complete definition of the shape in 3D needed? 
* Required & provided areas
  + Would it be beneficial for you, if there would be possibility to input only the info about necessary/provided area instead of definition of real bars? 

Please, let us know, what you think about it. We appreciate your feedback.

Thank you all in advance!

**Comment by DL-Lukas** [comment]
Hi,
we (Dlubal) went through the proposal and have following ideas. The first one (Count of legs in y-direction) is important for columns and probably also seimic designs. We think it would be in general important to support it. The other 2 options are a bit more exotic.
We are also working on the implementation of these features. So they are not ready in RFEM6, but will be in future. 

Stirrups:

1) Count of legs in y-direction (important)
Right now the proposal only allows the definition of multiple legs in z-direction. If I am not mistaken, we have no option to define more legs in y-direction. I have following setup in mind:
![image](https://user-images.githubusercontent.com/119600600/205061513-736b965e-bea2-431e-8242-e92013bcf74f.png)

2) Different stirrup-diameter in the same section of the beam. 

3) Diagonal stirrups (more common for seismic design situations)
![image](https://user-images.githubusercontent.com/119600600/205068279-984ac087-5ec6-4930-9c0d-8d831a6cf44b.png)

We understand the general idea to use it as "virtual reinforcement sketch". But we are interested in sharing as many details as possible related to the reinforcement. Therefor we favor the idea of sharing also anchroage related information.

Best Regards
Lukas

**Reply by robinblazek** [reply]
Hi @DL-Lukas , first of all thank you very much for your input! 

Ad 1) Very good remark. You're right, based on the current proposal, there is no possibility to influence count of legs in the other direction. You could possibly create it with some specifically defined big "cover" property, from one or another side. But possibility to have more legs in other direction sounds nice and straight forward. We evaluate it and possibly include this into our proposal. 

Ad 2) Do you mean for example different diameters for the yellow highlighted stirrups in the picture from you (point 1)? We will evaluate this possibility. I believe it could be handled. 

Ad 3) This is a little bit more advanced. I believe that these could be handled via some of the "general description of the geometry" (for example similar as CompositeShapDef polygons, as we mentioned). 

**Thanks for remarks and ideas, we really appreciate it!**  We will try to evaluate them and possibly include them to the proposal. 

Best regards,
Robin

**Reply by DL-Lukas** [reply]
Hey @robinblazek ,
sorry for the late reply. 

About 2) Correct. For example different diameters for the yellow stirrups in my picture.

**Comment by CONC-Dan** [comment]
Hi, for our experience (Concrete.it) defining type, position, characteristics of the reinforcement, for each type of possible section, is a very complex undertaking.
In other export types (e.g. IFC, Revit) we prefer to export:
- material
- diameter
- the type of bar (bar, stirrup, strand, etc.)
- the geometry of each bar, as a 3D polyline (including anchors)
- the host element
Other properties, such as positioning and quantities, can be derived from this data.
For these reasons it seems easier to implement a general reinforcement type, based on a point definition (as in CompositeShapeDef object in SAF).
This could be accompanied by specific types, parametric, for the simplest cases to manage.
